provider/image.go

```diff
@@ -428,6 +438,43 @@ func getCredentials() (map[string]clitypes.AuthConfig, error) {
 	return auths, nil
 }
 
+// Because the authConfigs provided by the host include the `https://` prefix in the map keys, `getRegistryAddrForAuth`
+// ensures we return a registry address that includes the `https://` scheme.
+// While this prefix is not needed for program-provided auth, it is valid regardless, so adding it by default
+// keeps the special case handling to a minimum.
+func getRegistryAddrForAuth(serverName, imgName string) (string, error) {
+	var serverAddr string
+	if serverName == "docker.io" {
+		// if it's dockerhub, we special case it so host config can find the correct registry
+		return "https://index.docker.io/v1/", nil
+	}
+
+	if serverName == "" {
+		// if there is no servername in the registry input, we attempt to build it from the fully qualified image name.
+		addr, _, found := strings.Cut(imgName, "/")
+		if !found {
+			return "", errors.Errorf("image name must be fully qualified: %s", imgName)
```

Comment:
`errors` has been deprecated in favor of the std lib.
```suggestion
			return "", fmt.Errorf("image name must be fully qualified: %s", imgName)
```